src/Raven.Server/NotificationCenter/Handlers/DatabaseNotificationCenterHandler.cs

```diff
@@ -6,11 +6,56 @@
 using Raven.Server.Documents;
 using Raven.Server.NotificationCenter.Notifications;
 using Raven.Server.Routing;
+using Raven.Server.ServerWide.Context;
+using Sparrow.Json;
 
 namespace Raven.Server.NotificationCenter.Handlers
 {
     public class DatabaseNotificationCenterHandler : DatabaseRequestHandler
     {
+        [RavenAction("/databases/*/notification-center/get", "GET", AuthorizationStatus.ValidUser, EndpointType.Read, SkipUsagesCount = true)]
+        public async Task GetNotifications()
+        {
+            var postponed = GetBoolValueQueryString("postponed", required: false) ?? true;
+            var type = GetStringQueryString("type", false);
+            var shouldFilter = type != null;
+            if (shouldFilter && Enum.TryParse(typeof(NotificationType), type, out _) == false)
+                throw new ArgumentException($"The 'type' parameter must be a type of '{{nameof(NotificationType)}}'. Instead, got '{type}'.");
+            
+            using (ContextPool.AllocateOperationContext(out DocumentsOperationContext context))
+            await using (var writer = new AsyncBlittableJsonTextWriter(context, ResponseBodyStream()))
+            using (Database.NotificationCenter.GetStored(out var storedNotifications, postponed))
+            {
+                writer.WriteStartObject();
+                writer.WritePropertyName("Results");
+                writer.WriteStartArray();
+                var isFirst = true;
+                foreach (var notification in storedNotifications)
+                {
+                    if (shouldFilter && notification.Json != null)
+                    {
+                        if (notification.Json.TryGet(nameof(Notification.Type), out string notificationType) == false)
+                            continue;
+                        
+                        if (notificationType != type)
+                            continue;
+                    }
+                    
+                    
+                    if (isFirst == false)
+                    {
+                        writer.WriteComma();
+                    }
+                    
+                    writer.WriteObject(notification.Json);
+                    isFirst = false;
+                }
+                
+                writer.WriteEndArray();
```

Comment:
I'm missing paging, would love to have sth like TotalCount or TotalResults and even in-memory paging here with optimization that pageSize=0 will just return count, like we have for regular queries

Reply:
Done :)